onmt/data/SampledDataset.lua

```diff
@@ -0,0 +1,266 @@
+--[[ Data management and batch creation. Handles data created by `preprocess.lua`. ]]
+require 'torchx'
```

Comment:
OK. Let's put it in the `onmt/data` directory then with the license in the file header.